src/extension/executors/runner.ts

```diff
@@ -119,15 +119,31 @@ export async function executeRunner(
     program.handleInput(input)
   })
 
-  const useIntegrated = isRunmeIntegratedTerminalEnabled()
+  const revealNotebookTerminal = interactive ?
+    background ?
+      isNotebookTerminalFeatureEnabled('backgroundTask') :
+      isNotebookTerminalFeatureEnabled('interactive') :
+    isNotebookTerminalFeatureEnabled('nonInteractive')
 
   if (!interactive) {
     const output: Buffer[] = []
 
     const mime = mimeType || 'text/plain' as const
 
+    if (revealNotebookTerminal) {
+      await replaceOutput(exec, [
+        new NotebookCellOutput([
+          NotebookCellOutputItem.json(NotebookTerminal.create(cellUUID), OutputType.terminal),
+        ]),
+      ])
+    }
+
     // adapted from `shellExecutor` in `shell.ts`
     const handleOutput = async (data: Uint8Array) => {
+      if (isNotebookTerminalFeatureEnabled('nonInteractive')) {
```

Comment:
As was mentioned in the previous comment, you can use `revealNotebookTerminal` instead of calling the function

Reply:
Good catch